modules/blender/hm.nix

```
@@ -0,0 +1,42 @@
+{
+  config,
+  lib,
+  ...
+}:
+let
+  blenderVersions = [
+    "blender/4.2/scripts/presets/interface_theme"
+    "blender/4.3/scripts/presets/interface_theme"
+    "blender/4.4/scripts/presets/interface_theme"
+  ];
+in
+{
+  options.stylix.targets.blender.enable =
+    config.lib.stylix.mkEnableTarget "blender" true;
+
+  config =
+    lib.mkIf (config.stylix.enable && config.stylix.targets.blender.enable)
+      {
+        xdg.configFile = builtins.listToAttrs (
+          builtins.concatMap (version: [
+            {
+              name = "${version}/Stylix.xml";
+              value.text =
+                let
+                  coloredFile = config.lib.stylix.colors {
+                    template = ./Stylix.xml.mustache;
+                    extension = ".xml";
+                  };
+                in
+                builtins.replaceStrings
+                  [ "%POPUPSFONTSIZE%" "%DESKTOPFONTSIZE%" ]
+                  [
+                    (toString config.stylix.fonts.sizes.popups)
+                    (toString config.stylix.fonts.sizes.desktop)
+                  ]
+                  (builtins.readFile coloredFile);
+            }
+          ]) blenderVersions
+        );
```

What about declaring the actual versions:

```nix
let
  versions = [
    "4.2"
    "4.3"
    "4.4"
  ];
in
```

and caching the `builtins.readFile (config.lib.stylix.colors /* ... */)` result, and replacing `builtins.listToAttrs (builtins.concatMap /* ... */)` with `builtins.foldl'`:

```suggestion
        {
          xdg.configFile =
            let
              theme = builtins.readFile (
                config.lib.stylix.colors {
                  template = ./Stylix.xml.mustache;
                  extension = ".xml";
                }
              );
            in
            builtins.foldl' (
              acc: version:
              acc
              // {
                "blender/${version}/scripts/presets/interface_theme/Stylix.xml".text =
                  builtins.replaceStrings
                    [ "%POPUPSFONTSIZE%" "%DESKTOPFONTSIZE%" ]
                    [
                      (toString config.stylix.fonts.sizes.popups)
                      (toString config.stylix.fonts.sizes.desktop)
                    ]
                    theme;
              }
            ) { } versions;
        }
```